src/item/Item.js

```diff
@@ -1029,6 +1029,19 @@ new function() { // Injection scope for various item event handlers
             }
         },
 
+        /**
+         * Clears cached global matrix for item and all its descendants.
+         * This is used to avoid invalid coodrinates calculation (#1448).
+         */
+        _clearGlobalMatrixCache: function(item) {
+            item._globalMatrix = undefined;
+            if (item.children && item.children.length > 0) {
+                for (var i = 0, l = item.children.length; i < l; i++) {
+                    Item._clearGlobalMatrixCache(item.children[i]);
+                }
+            }
```

Comment:
Here you repeatedly access `item.children`. For performance and also code minification it is better to reference it in a local variable first:
`var children = item.children;`
Also, you don't need to check for `length` if  your next statement is a loop that only loops ifthe length is > 0, so:
```js
            var children = this.children;
            if (children) {
                for (var i = 0, l = children.length; i < l; i++) {
                    children[i]._clearGlobalMatrix();
                }
            }
   
 ```